algorithms/baselines/ppo/agent_ppo.py

```diff
@@ -27,10 +27,25 @@ def __init__(self, env, ph_observations, params):
         num_actions = env.action_space.n
         obs_space = env.observation_space
 
+        # Goal observation
+        self.ph_goal_obs = None
+        self.is_goal_env = is_goal_based_env(env)
+        if self.is_goal_env:
+            # goal obs has the same shape as main obs
+            self.ph_goal_obs = placeholder_from_space(main_observation_space(env))
+
+        make_encoder_func = make_encoder_with_goal if self.is_goal_env else make_encoder
```

Comment:
Just realized that you don't really need this because the encoder is created inside if/else anyway (because different number of parameters). This is just a small improvement